zkchannels-crypto/src/pedersen.rs

```diff
@@ -208,82 +208,52 @@ mod test {
     #[cfg(feature = "bincode")]
     use {rand::Rng, std::convert::TryFrom};
 
-    fn commit_open<G: Group<Scalar = Scalar>>() {
+    fn commit_open_success<G: Group<Scalar = Scalar>, const N: usize>() {
         let mut rng = crate::test::rng();
-        let params = PedersenParameters::<G, 3>::new(&mut rng);
+        let params = PedersenParameters::<G, N>::new(&mut rng);
         let msg = Message::random(&mut rng);
         let bf = BlindingFactor::new(&mut rng);
 
         let com = msg.commit(&params, bf);
         assert!(com.verify_opening(&params, bf, &msg));
     }
 
-    #[test]
-    fn commit_open_g1() {
-        commit_open::<G1Projective>()
-    }
-
-    #[test]
-    fn commit_open_g2() {
-        commit_open::<G2Projective>()
-    }
-
-    fn commit_does_not_open_on_wrong_msg<G: Group<Scalar = Scalar>>() {
+    fn commit_does_not_open_on_wrong_msg<G: Group<Scalar = Scalar>, const N: usize>() {
         let mut rng = crate::test::rng();
-        let params = PedersenParameters::<G, 3>::new(&mut rng);
+        let params = PedersenParameters::<G, N>::new(&mut rng);
         let msg = Message::random(&mut rng);
         let bf = BlindingFactor::new(&mut rng);
 
         let bad_msg = Message::random(&mut rng);
 
         assert_ne!(
             &*msg, &*bad_msg,
-            "unfortunate RNG seed: bad_msg should be different"
+            "unfortunate RNG seed in commit_does_not_open_on_wrong_msg: bad_msg should be different"
```

Comment:
If this test fails, we should get a stack trace that will tell us which function failed, so you don't need to put the name in the error message.

Reply:
Oh yeah, I meant to ask about that. Thanks!